addons/account_payment/controllers/portal.py

```diff
@@ -55,15 +60,20 @@ def _invoice_get_page_view_values(self, invoice, access_token, **kwargs):
                 providers_sudo
             ),
         }
+        next_installment = invoice.get_next_installment_due()
         payment_context = {
             'amount': invoice.amount_residual,
+            'amount_custom': float(kwargs['amount']) if kwargs.get('amount') else False,
+            'amount_next_installment': next_installment['amount_residual'] if next_installment else False,
+            'number_next_installment': next_installment['number'] if next_installment else False,
```

Comment:
Again, do all computations here and put the values controling the view instead. That way the view is dumb and just display things depending of what you give here